test/datetime.cc

```diff
@@ -583,9 +583,56 @@ TEST(DateTime, TestSecondOfWeek) {
   EXPECT_EQ(a, e) << "Wrong second of week";
 }
 
+TEST(DateTime, DiffCaching) {
+  // no cache
+  const auto& tzdb = DateTime::get_tz_db();
+  auto diff = DateTime::timezone_diff(1586660072, tzdb.from_index(110), tzdb.from_index(94));
+  EXPECT_EQ(diff, -3 * 60 * 60);
+
+  // for a really long route that crosses many timezone we end up doing a lot of tz diffing
+  // the tuple is: number of calls, origin tz index, destination tz index
+  // 0 is reserved for no timezone so we shift down by one to use them with the tzdb
+  std::vector<std::tuple<int, int, int>> test_cases = {
```

Comment:
then we take what is a play by play of all the calls to the timezone diff method for a very long route through africa starting in europe